### PR TITLE
packages/dtc: Add 1.6.0

### PR DESCRIPTION
**Pull request by elijahr**
My config with dtc 1.5.1 was failing to build with:

```
[DEBUG]    ==> Executing:  'make' 'all' 'CC=x86_64-build_pc-linux-gnu-gcc' 'AR=x86_64-build_pc-linux-gnu-ar' 'PREFIX=/home/ct-ng/configs/aarch64-archlinux-linux-gnu/.build/aarch64-unknown-linux-gnu/buildtools' 'PKG_CONFIG=/bin/true' 'NO_PYTHON=1' 'NO_YAML=1' 'BIN=dtc'
[ALL  ]    make[1]: Entering directory '/home/ct-ng/configs/aarch64-archlinux-linux-gnu/.build/aarch64-unknown-linux-gnu/build/build-dtc-build'
...
[ALL  ]    	 LD dtc
[ALL  ]    /usr/sbin/ld: dtc-parser.tab.o:/home/ct-ng/configs/aarch64-archlinux-linux-gnu/.build/aarch64-unknown-linux-gnu/build/build-dtc-build/dtc-parser.tab.c:1089: multiple definition of `yylloc'; dtc-lexer.lex.o:/home/ct-ng/configs/aarch64-archlinux-linux-gnu/.build/aarch64-unknown-linux-gnu/build/build-dtc-build/dtc-lexer.l:26: first defined here
[ERROR]    collect2: error: ld returned 1 exit status
[ERROR]    make[1]: *** [Makefile:335: dtc] Error 1
[ALL  ]    make[1]: Leaving directory '/home/ct-ng/configs/aarch64-archlinux-linux-gnu/.build/aarch64-unknown-linux-gnu/build/build-dtc-build'
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing dtc for build'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@376]
[ERROR]  >>        called from: do_dtc_backend[scripts/build/companion_tools/500-dtc.sh@81]
[ERROR]  >>        called from: do_companion_tools_dtc_for_build[scripts/build/companion_tools/500-dtc.sh@18]
[ERROR]  >>        called from: do_companion_tools_for_build[scripts/build/companion_tools.sh@35]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@695]
```

The key error being `multiple definition of yylloc`. 

I did some searching and found that other projects had encountered a similar [issue](https://review.lineageos.org/c/LineageOS/android_kernel_oneplus_sm8150/+/273023).

dtc 1.6.0 fixed this by [removing the duplicate yylloc](https://github.com/dgibson/dtc/commit/0e9225eb0dfec51def612b928d2f1836b092bc7e).

~~Also this PR fixes a typo in the README.~~ Typo fix moved to https://github.com/crosstool-ng/crosstool-ng/pull/1445

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ git checkout -b fix_out_of_date_patch origin/1.22
 ```
  Then when you get to this pull request screen change the base branch from `master` to `1.22`
 
-- By creating a pull request, the PR is entered into the [backlog](https://waffle.io/crosstool-ng/crosstool-ng). A [travis-ci](https://travis-ci.org/crosstool-ng/crosstool-ng/builds) job will run to test your changes against a select set of samples. As they start to get worked, they should be placed in the `Ready` state. PRs that are being worked are `In Progress`. If a questions come up about the commit that might envolve changes to the commit then the PR is placed in `Waiting For Response`, you have two options:
+- By creating a pull request, the PR is entered into the [backlog](https://waffle.io/crosstool-ng/crosstool-ng). A [travis-ci](https://travis-ci.org/crosstool-ng/crosstool-ng/builds) job will run to test your changes against a select set of samples. As they start to get worked, they should be placed in the `Ready` state. PRs that are being worked are `In Progress`. If a questions come up about the commit that might involve changes to the commit then the PR is placed in `Waiting For Response`, you have two options:
```
Comment (cpackham) [inline README.md:93]: This change should be in a separate commit as it is unrelated to the dtc change

File: packages/dtc/1.6.0/chksum [hunk 2]
```diff
@@ -0,0 +1,8 @@
+md5 dtc-1.6.0.tar.xz 1556ba93648bf70d7aa034252e278751
+sha1 dtc-1.6.0.tar.xz 3e6ab5121dcfe5ff9f8b0503becfcf687d5d4434
+sha256 dtc-1.6.0.tar.xz 10503b0217e1b07933e29e8d347a00015b2431bea5f59afe0bed3af30340c82d
+sha512 dtc-1.6.0.tar.xz 15433b24f1d6b7ed1f8066d050bd1bcbf988731aa38147564e3dd04b5f69af8d69e03befdc621a768526c620425a9bdd24aad4f9ff135930d6a1eeb0625f7de3
+md5 dtc-1.6.0.tar.gz b87b7aee4a01535ef1f9c53d5f6cd178
+sha1 dtc-1.6.0.tar.gz ef037f4599b7b1e883c9bb588b24d289dc9cd1f2
+sha256 dtc-1.6.0.tar.gz 9fbe07223a98f2d7088a340b5505d4dfe682d77580e788d08cfcc1b61d8be237
+sha512 dtc-1.6.0.tar.gz 8ca3f9e06a60144d3f476c251d911f2daa16afb82daa101c61050bfab226dbb330f8a75a3715fe0c2b46b2a64f3794eb3ce0816aef434ab64e1feb1ba08be574
```
Comment (cpackham) [inline packages/dtc/1.6.0/chksum:8]: These match what I calculate

**Review by cpackham — CHANGES_REQUESTED** [review]
Looks pretty good just need to split off the README changes

File: README.md (hunk 1 above)
Comment (elijahr) [reply README.md]: @cpackham I've rebased on master and removed this change. Will open a separate PR for the README change.

Comment (elijahr) [reply README.md]: Opened https://github.com/crosstool-ng/crosstool-ng/pull/1445

**Review by cpackham — APPROVED** [review]
